File: 3936-split-array-by-prime-indices/3936-split-array-by-prime-indices.py
class Solution(object):
    def splitArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        def isPrime(num):
            if num <= 1:
                return False
            for i in range(2, int(num ** 0.5) + 1):
                if num % i == 0:
                    return False
            return True

        n = len(nums)
        sumA, sumB = 0, 0
        for i in range(n):
            if isPrime(i):
                sumA += nums[i]
            else:
                sumB += nums[i]
        return abs(sumA - sumB)

File: 3936-split-array-by-prime-indices/3936-split-array-by-prime-indices.c
bool isPrime(int num) {
    if (num <= 1)
        return false;
    for (int i = 2; i <= sqrt(num); ++i) {
        if (num % i == 0)
            return false;
    }
    return true;
}

long long splitArray(int* nums, int numsSize) {
    long long sumA = 0, sumB = 0;
    for (int i = 0; i < numsSize; i++) {
        if (isPrime(i))
            sumA += nums[i];
        else
            sumB += nums[i];
    }
    return fabs(sumA - sumB);
}